Add support for 400s PM 2.5 readings

bdwyer2 opened the pull request:
issue #134

bdwyer2 committed:
Add support for 400s PM 2.5 readings

bdwyer2 committed:
fix flake8

bdwyer2 committed:
Typo

webdjoe commented:
@bdwyer2 Thank you for the PR. Can you please make a child class for the 400s with the altered `get_details()` method and `air_quality` property so that the methods and properties remain consistent across all models

alexanv1 commented:
@webdjoe, it looks like air_quality property used to actually return ```air_quality_value``` but this regressed as part of commit https://github.com/webdjoe/pyvesync/commit/1aae09bc4b5ab6ba69bcf1fd96fe58f5bed2deec and now air_quality property returns ```air_quality``` (which seems to be an enum of some sort).

Was the change intentional? If not, would it make sense to go back to the previous behavior since air_quality property is still documented as returning PM2.5?

alexanv1 commented on src/pyvesync/vesyncfan.py:
```diff
@@ -184,6 +184,8 @@ def build_purifier_dict(self, dev_dict: dict) -> None:
         else:
             self.device_status = 'off'
         self.details['filter_life'] = dev_dict.get('filter_life', 0)
+        self.details['air_quality_value'] = dev_dict.get('air_quality_value',
```
I think this belongs under ```if self.air_quality_feature:``` block below since it applies only to those Air Purifiers that support this (Core300S and Core400s support it, Core200s does not)

alexanv1 commented:
@bdwyer2, could you also please fix what looks like an existing typo:
```ait_quality``` in the code should be ```air_quality``` I think

Thanks!

webdjoe commented:
@bdwyer2 thank you for pointing that out, it looks like my error when refactoring. So, it would be safe to just rename the get attribute to `air_quality_value`

bdwyer2 commented:
I only have a 300s and 400s to test on. Do any purifiers use `air_quality` or do they all use `air_quality_value`?

alexanv1 commented:
> I only have a 300s and 400s to test on. Do any purifiers use `air_quality` or do they all use `air_quality_value`?

I also only have 300s and 400s so can't comment on other models but changing to `air_quality` was an accidental regression it seems so it should be safe to go back to `air_quality_value` and any issues for other models (which would have already existed) can be handled separately.

webdjoe committed:
Fix air_quality and air_quality_value

Differentiate air_quality as a level of air quality and air_quality_value as the PM2.5 particulate reading

webdjoe committed:
version bump

webdjoe commented:
`air_quality` is the level assigned by vesync for rating the air quality and `air_quality_value` is the particulate measurement value. I fixed the pr. Will release shortly

bdwyer2 commented:
Thanks for finishing the PR @webdjoe

alexanv1 commented:
@webdjoe, would it be possible to get a new pyvesync release with this fix?
Thanks!
